json-rpc/src/stream_rpc/tests/util.rs

```diff
@@ -0,0 +1,142 @@
+    queue_size: usize,
+    poll_interval_ms: u64,
+) -> (Arc<MockDiemDB>, StreamConfig) {
+    diem_logger::DiemLogger::init_for_testing();
+    let db = Arc::new(mock_db());
+    let config = make_ws_config(fetch_size, queue_size, poll_interval_ms);
+    (db, config)
+}
+
+/// Verifies the subscription "OK" response is correct
+pub fn verify_ok(msg: Message, name: &str) -> u64 {
+    let msg_str = msg
+        .to_str()
+        .unwrap_or_else(|_| panic!("{}: ok msg response", &name));
+
+    let resp: JsonRpcResponse = serde_json::from_str(msg_str)
+        .unwrap_or_else(|_| panic!("{}: could not parse response. {:?}", &name, msg));
+
+    let res = resp.result;
+    assert!(res.is_some(), "{}: result is None. {:?}", &name, msg);
+
+    let res = res.unwrap();
+
+    let status = res.get("status");
+    assert!(status.is_some(), "{}: status is None. {:?}", &name, msg);
+
+    assert_eq!(
+        status.unwrap().as_str().unwrap(),
+        "OK",
+        "{}: status is not 'OK'. {:?}",
+        &name,
+        msg
+    );
+
+    res.get("transaction_version")
+        .expect("OK transaction version")
+        .as_u64()
+        .expect("ok transaction version to u64")
+}
+
+pub fn num_clients(cm: &ConnectionManager) -> usize {
+    cm.clients.read().len()
+}
+
+pub fn get_latest_client(cm: &ConnectionManager) -> ClientConnection {
+    cm.get_client((num_clients(cm) - 1) as u64)
+        .expect("no clients exist")
+}
+
+pub fn num_tasks(client: &ClientConnection) -> usize {
+    client.tasks.lock().len()
+}
+
+/// This is primarily to ensure that the websocket transport responds to `close` correctly
+/// And passes the appropriate signal along to the `ConnectionManager`
+/// Dropping the ws_client is enough to actually close the connection
+pub async fn close_ws(mut ws_client: WsClient, name: &str) {
+    ws_client.send(warp::ws::Message::close()).await;
+    tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
+    ws_client
+        .recv_closed()
+        .await
+        .unwrap_or_else(|err| panic!("{}: connection not closed. {:?}", name, err));
+
+    tokio::time::sleep(tokio::time::Duration::from_millis(200)).await;
+}
+
+pub async fn next_message(ws_client: &mut WsClient, name: &str) -> warp::filters::ws::Message {
+    timeout(100, ws_client.recv(), name)
+        .await
+        .unwrap_or_else(|e| panic!("{}: message not ok. {:?}", name, e))
+}
+
+pub async fn timeout<F: Future<Output = T>, T>(duration_millis: u64, future: F, name: &str) -> T {
+    tokio::time::timeout(std::time::Duration::from_millis(duration_millis), future)
+        .await
+        .unwrap_or_else(|_| panic!("{}: Timed out", name))
+}
+
+pub fn sync_async<T, Fut>(handle: &Handle, f: Fut) -> T
+where
+    Fut: Future<Output = T> + Send + 'static,
+    T: Send + 'static,
+{
+    let (tx, rx) = sync_channel(1);
+    handle.spawn(async move {
+        let res = f.await;
+        let _ = tx.send(res);
+    });
+    rx.recv().unwrap()
```

Comment:
You should simplify this function by using [`Handle::block_on`](https://docs.rs/tokio/1.6.1/tokio/runtime/struct.Handle.html#method.block_on)